spec/std/json/any_spec.cr

```diff
@@ -0,0 +1,92 @@
+require "spec"
+require "json"
+
+describe JSON::Any do
+  describe "casts" do
+    it "gets nil" do
+      JSON.parse("null").as_nil.should be_nil
+    end
+
+    it "gets bool" do
+      JSON.parse("true").as_bool.should be_true
+      JSON.parse("false").as_bool.should be_false
+    end
+
+    it "gets int" do
+      JSON.parse("123").as_i.should eq(123)
+      JSON.parse("123456789123456").as_i64.should eq(123456789123456)
+    end
+
+    it "gets float" do
+      JSON.parse("123.45").as_f.should eq(123.45)
+      JSON.parse("123.45").as_f32.should eq(123.45_f32)
+    end
+
+    it "gets string" do
+      JSON.parse(%("hello")).as_s.should eq("hello")
+    end
+
+    it "gets array" do
+      JSON.parse(%([1, 2, 3])).as_a.should eq([1, 2, 3])
```

Comment:
Oh I see.

For sure not in a scope of this PR: Could it be possible to make a thin wrapper `JSON::AnyArray`, that will wrap `Array(JSON::Type)` and will provide neat interface, like `#[], #[]?, #each` that will on the fly re-wrap values of type `JSON::Type` in `JSON::Any` ?